R/plotting-functions.R

```diff
@@ -0,0 +1,223 @@
+#'     labelPos = 2, labelOffset = 0.1, labelSrt = -30)
+#'
+#' ## Add a custom annotation for each peak
+#' sp$label <- list(c("", "A", "B", "C", "D"),
+#'     c("Frodo", "Bilbo", "Peregrin", "Samwise"))
+#' ## Plot each peak in a different color
+#' plotSpectra(sp, labels = function(z) unlist(z$label),
+#'     col = list(1:5, 1:4))
+#'
+#' ## Plot a single spectrum specifying the label
+#' plotSpectra(sp[2], labels = c("A", "B", "C", "D"))
+NULL
+
+#' @rdname spectra-plotting
+#'
+#' @importFrom graphics par
+#'
+#' @export plotSpectra
+plotSpectra <- function(x, xlab = "m/z", ylab = "intensity",
+                        type = "h", xlim = numeric(),
+                        ylim = numeric(), main = paste("RT", rtime(x)),
+                        col = "#00000080", labels = character(),
+                        labelCex = 1, labelSrt = 0,
+                        labelAdj = NULL, labelPos = NULL,
+                        labelOffset = 0.5, ...) {
+    nsp <- length(x)
+    if (nsp == 1)
+        col <- list(col)
+    if (length(col) != nsp)
+        col <- rep(col[1], nsp)
+    if (length(main) != nsp)
+        main <- rep(main[1], nsp)
+    par(mfrow = c(round(sqrt(nsp)), ceiling(sqrt(nsp))))
+    for (i in seq_len(nsp))
+        .plot_single_spectrum(x[i], xlab = xlab, ylab = ylab, type = type,
+                              xlim = xlim, ylim = ylim, main = main[i],
+                              col = col[[i]], labels = labels,
+                              labelCex = labelCex, labelSrt = labelSrt,
+                              labelAdj = labelAdj, labelPos = labelPos,
+                              labelOffset = labelOffset, ...)
+}
+
+#' @description
+#'
+#' Plot a single spectrum (m/z on x against intensity on y) with the optional
+#' possibility to label the individual peaks.
+#'
+#' @author Johannes Rainer, Sebastian Gibb
+#'
+#' @importFrom graphics axis box plot.new plot.window plot.xy strwidth text title
+#'
+#' @importFrom grDevices dev.flush dev.hold xy.coords
+#'
+#' @examples
+#'
+#' ints <- c(4.3412, 12, 8, 34, 23.4)
+#' mzs <- c(13.453421, 43.433122, 46.6653553, 129.111212, 322.24432)
+#'
+#' df <- DataFrame(msLevel = 1L, rtime = 123.12)
+#' df$mz <- list(mzs)
+#' df$intensity <- list(ints)
+#' sp <- Spectra(df)
+#'
+#' .plot_single_spectrum(sp, main = "hello")
+#' .plot_single_spectrum(sp, bty = "n")
+#' .plot_single_spectrum(sp, frame.plot = FALSE)
+#'
+#' .plot_single_spectrum(sp, col = 1:5)
+#'
+#' .plot_single_spectrum(sp, labels = 1:5, col = 1:5)
+#'
+#' .plot_single_spectrum(sp, labels = format(mz(sp)[[1]], digits = 5),
+#'     labelPos = 2, labelOffset = 0.1, labelSrt = -30)
+#' grid()
+#' .plot_single_spectrum(sp, col = "red", type = "p", add = TRUE)
+#'
+#' .plot_single_spectrum(sp, labels = function(z) format(mz(z)[[1]], digits = 5),
+#'     labelPos = 2, labelOffset = 0.1, labelSrt = -30)
+#' grid()
+#'
+#' @noRd
+.plot_single_spectrum <- function(x, xlab = "m/z", ylab = "intensity",
```

Comment:
Is this function the same as `MSnbase:::.plotSingleSpectrum()`? Is so, may be mention is in the source code and/or man page.

Reply:
Actually no - it is a more simple (I guess) function to simply draw a spectrum and have peaks eventually annotated. I was not even aware of the `MSnbase:::.plotSingleSpectrum` function.

Reply:
OK.